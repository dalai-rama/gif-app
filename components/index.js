

export * from './AddCategory'
export * from './DivGrid'
export * from './GifItem'
