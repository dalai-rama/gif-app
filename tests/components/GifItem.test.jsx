import { screen, render } from "@testing-library/react"
import { GifItem } from "../../components"



describe("Pruebas en <GifItem />", ()=>{

    const title = "Titulo";
    const url = "https://titulo.com/imagen.jpg";

    test("debe hacer match con el snapshot", ()=> {
        
        const {container} = render( <GifItem title={title} url={url} /> );
        expect({container}).toMatchSnapshot();

    })

    test('debe mostrar la imagen con el URL y ALT indicado', ()=>{

        render( <GifItem title={title} url={url} /> );
        const { src, alt } = screen.getByRole('img');
    
        expect ( src ).toBe( url );
        expect ( alt ).toBe( alt );

    })

    test('debe tener titulo', ()=>{
        
        render( <GifItem title={title} url={url} /> );
        expect(screen.getByText( title)).toBeTruthy();
        
    })

})