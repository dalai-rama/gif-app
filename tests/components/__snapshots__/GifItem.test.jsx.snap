// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
Object {
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="Titulo"
        src="https://titulo.com/imagen.jpg"
      />
      <p>
         
        Titulo
         
      </p>
    </div>
  </div>,
}
`;
